src/fhirtypes/common.ts

```diff
@@ -855,14 +855,12 @@ export function replaceReferences<T extends AssignmentRule | CaretValueRule>(
       }
     }
   } else if (value instanceof FshCode) {
-    const [system, ...versionParts] = value.system?.split('|') ?? [];
-    const version = versionParts.join('|');
-    const codeSystem = tank.fish(system, Type.CodeSystem);
-    const codeSystemMeta = fisher.fishForMetadata(system, Type.CodeSystem);
+    const codeSystem = fishInTankBestVersion(tank, value.system, Type.CodeSystem);
+    const codeSystemMeta = fishForMetadataBestVersion(fisher, value.system, Type.CodeSystem);
```

Comment:
This is not really a problem, but it's possible for each of these functions to separately log a warning for a version mismatch if the CodeSystem is retrieved from the package when fishing for metadata. Here's an example that can make this happen:
```
CodeSystem: BreakfastCS
* ^version = "1.0.0"
* #toast "Toast"
* #donut "Donut"

Instance: MyBreakfast
InstanceOf: Observation
* status = #final
* code = BreakfastCS|2.0.0#toast
```

Reply:
Ah, yeah. We technically need to do both fishes -- so the only way I can think of to avoid the double warning is to do the 2nd one based on whatever was actually returned by the first (rather than the original request). E.g.,:

```ts
    const codeSystemMeta = fishForMetadataBestVersion(fisher, value.system, Type.CodeSystem);
    if (codeSystemMeta) {
      clone = cloneDeep(rule);
      const assignedCode = getRuleValue(clone) as FshCode;
      assignedCode.system = value.system.replace(/^[^|]+/, codeSystemMeta.url);

      // Find the code system using the returned metadata to avoid duplicate warnings if version mismatches
      const matchedCanonical = codeSystemMeta.url
        ? `${codeSystemMeta.url}${codeSystemMeta.version ? `|${codeSystemMeta.version}` : ''}`
        : value.system;
      const codeSystem = fishInTankBestVersion(tank, matchedCanonical, Type.CodeSystem);
      if (codeSystem && (codeSystem instanceof FshCodeSystem || codeSystem instanceof Instance)) {
        // if a local system was used, check to make sure the code is actually in that system
        listUndefinedLocalCodes(codeSystem, [assignedCode.code], tank, rule);
      }
    }
```

@mint-thompson - I have this change locally and it _does_ worth. Do you think I should push it up?

Reply:
I think so, yes. It's not too complicated, and it makes the output more consistent. Thank you!

Reply:
OK. Fixed in 858c75d. I also noticed that the warning messages did not log the source info -- so we'd say there was a mismatch, but it would be up to the author to find it in their FSH files. This didn't seem optimal, so I also updated it to pass along the source info and log that. It's in the same commit.